client/src/components/OneLineForm/OneLineForm.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, useState } from "react";
+import styles from "./OneLineForm.module.css";
+
+interface Props extends React.ComponentPropsWithoutRef<"input"> {
+  inputRef?: (node: HTMLInputElement | null) => void;
+  onFormSubmit: (value: string) => Promise<void>;
+}
+
+const OneLineForm: FC<Props> = ({
+  onFormSubmit,
+  inputRef,
+  ...inputAttributes
+}) => {
+  const [value, setValue] = useState("");
+  return (
+    <form
+      onSubmit={async (e) => {
+        e.preventDefault();
+        if (value === "") {
+          return;
+        }
+        await onFormSubmit(value);
+        setValue("");
+      }}
+      className={styles.form}
+    >
+      <input
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
```

Comment:
from react docs
_"Using an arrow function in render creates a new function each time the component renders, which may break optimizations based on strict identity comparison."_
[source](https://reactjs.org/docs/faq-functions.html)

Reply:
It looks like they say it's ok and it might be a slight performance penalty
https://reactjs.org/docs/faq-functions.html#is-it-ok-to-use-arrow-functions-in-render-methods